src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -524,35 +553,36 @@ private void processCloudWatchAttemptLogInformation(Map<String, Set<LogFile>> co
         // If we have completely read the file, then we need add it to the completed files list and remove it
         // it (if necessary) for the current processing list.
         String componentName = attemptLogInformation.getComponentName();
-        if (!logFileGroup.isActiveFile(file) && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
-                + cloudWatchAttemptLogFileInformation.getStartPosition()) {
+        long bytesUploaded = cloudWatchAttemptLogFileInformation.getBytesRead()
+                + cloudWatchAttemptLogFileInformation.getStartPosition();
+
+        if (!logFileGroup.isActiveFile(file) && file.length() == bytesUploaded) {
             Set<LogFile> completedFiles = completedLogFilePerComponent.getOrDefault(componentName,
                     new HashSet<>());
             completedFiles.add(file);
             completedLogFilePerComponent.put(componentName, completedFiles);
-            if (currentProcessingLogFilePerComponent.containsKey(componentName)) {
-                CurrentProcessingFileInformation fileInformation = currentProcessingLogFilePerComponent
-                        .get(componentName);
-                if (fileInformation.fileHash.equals(fileHash)) {
-                    currentProcessingLogFilePerComponent.remove(componentName);
-                }
-            }
-        } else {
-            // Add the file to the current processing list for the component.
-            // Note: There should always be only 1 file which will be in progress at any given time.
-            // @deprecated The fileName is deprecated value in versions greater than 2.2, but keep it here to avoid
-            // upgrade-downgrade issues.
-            CurrentProcessingFileInformation processingFileInformation =
-                    CurrentProcessingFileInformation.builder()
-                            .fileName(file.getSourcePath())
-                            .startPosition(cloudWatchAttemptLogFileInformation.getStartPosition()
-                                    + cloudWatchAttemptLogFileInformation.getBytesRead())
-                            .lastModifiedTime(cloudWatchAttemptLogFileInformation.getLastModifiedTime())
-                            .fileHash(fileHash)
-                            .build();
-            currentProcessingLogFilePerComponent.put(attemptLogInformation.getComponentName(),
-                    processingFileInformation);
         }
+
+
+        // Track each file that got uploaded. So that if modified in the future we can still track upload
+        // the new contents starting from the last upload position.
+
+        CurrentProcessingFileInformation processingFileInformation =
```

Comment:
We are now storing / tracking every single file per component type. Each component we track has an LRU whose capacity defaults to the max of 5 or the number of files that it processed. What this means is that we could track by default up 5 "active files per component/configuration". We can get smarter and dynamically resize the lru

Reply:
5 is very small if we're not dynamically resizing. I think that needs to be a P0 requirement.

Consider the case where the customer has one regex for all of their components, assuming the follow our recommended naming convention (`com.company.XYZ`)

Reply:
I guess so, let me try to add something here to handle this with a few tests

Reply:
What happens if one of the active logs doesn't get written to between LM iterations? It won't get returned in the log group, which could result in the hash being removed. I think this approach still has some fundamental issues

Reply:
Added a method below called adjust capacity which will increase the LR capacity such that it always stores the max amount of logFileGroup it has recorded. I don't have enough time but we can modify the adjust capacity method later to clear entries if the capacity number is lower after and a record has not been updated in say 2 days to clear some memory

Reply:
PS: I know that it won't be perfect unless we store every single file in existence for example if one cycle we get 6 files and then the next cycle we get other 6 completely different files the sure, we will have duplicates.

We could make it so it just keep adding elements forever and when a new element get added it evicts anything that has not been updated in say 2 days as well, but the time to eviction should be something the user should be able to configure

Reply:
I made some changes to the LRU to work based on time. File information that gets stored that has not been modified in the last MAX_CACHE_INACTIVE_TIME_SECONDS gets removed when a new record gets added. We could make it configurable later on but we could start with a sensible MAX_CACHE_INACTIVE_TIME_SECONDS number.

Reply:
Chatted with Joe about this offline